client/src/pages/Achieves/utils.ts

```diff
@@ -0,0 +1,168 @@
+import { GameCategories } from "../../interfaces/Categories";
+import { GameAttempt, User } from "../../interfaces/User";
+import { AchievesId } from "./types";
+
+export const checkIsAchieveOpen = (user: User | null, achieveId: AchievesId) => {
+  if (user) {
+    const stats = user.statistics;
+
+    switch (achieveId) {
+      case AchievesId.FRIENDS:
+        return checkFriendAchieve(user);
+
+      case AchievesId.DISCOVERER:
+        return checkDiscovererAchieve(stats);
+
+      case AchievesId.EXPLORER:
+        return checkExplorerAchieve(stats);
+
+      case AchievesId.SHELDON:
+        if (countMemoryAttemts(stats) >= 10) {
```

Comment:
Wouldn't it be better to return just 'countMemoryAttemts(stats) >= 10' instead of if statements?

Reply:
Refactored